avm/ptn/app-service-lza/hosting-environment/modules/app-service/app-service.module.bicep

```diff
@@ -0,0 +1,332 @@
+param zoneRedundant bool = true
+
+@description('Optional. Location for all resources.')
+param location string
+
+@description('Resource tags that we might need to add to all resources (i.e. Environment, Cost center, application name etc)')
+param tags object
+
+@description('Default is empty. If empty no Private Endpoint will be created for the resoure. Otherwise, the subnet where the private endpoint will be attached to')
+param subnetPrivateEndpointResourceId string = ''
+
+@description('Optional. Array of custom objects describing vNet links of the DNS zone. Each object should contain vnetName, vnetId, registrationEnabled')
+param virtualNetworkLinks array = []
+
+@description('Kind of server OS of the App Service Plan')
+@allowed(['windows', 'linux'])
+param webAppBaseOs string
+
+@description('An existing Log Analytics WS Id for creating app Insights, diagnostics etc.')
+param logAnalyticsWorkspaceResourceId string
+
+@description('The subnet ID that is dedicated to Web Server, for Vnet Injection of the web app. If deployAseV3=true then this is the subnet dedicated to the ASE v3')
+param subnetIdForVnetInjection string
+
+@description('Optional. If true, apps assigned to this App Service plan can be scaled independently. If false, apps assigned to this App Service plan will scale to all instances of the plan.')
+param perSiteScaling bool = false
+
+@description('Optional, default is 20. Maximum number of total workers allowed for this ElasticScaleEnabled App Service Plan.')
+param maximumElasticWorkerCount int = 20
+
+@description('Optional. Scaling worker count.')
+param targetWorkerCount int = 0
+
+@description('Optional, default is Windows. Kind of server OS.')
+@allowed([
+  'Windows'
+  'Linux'
+])
+param serverOS string = 'Windows'
+
+@description('Optional. The instance size of the hosting plan (small, medium, or large).')
+@allowed([
+  0
+  1
+  2
+])
+param targetWorkerSize int = 0
+
+@description('Optional. The site configuration for the web app.')
+param siteConfig object = {
+  alwaysOn: true
+  ftpsState: 'FtpsOnly'
+  minTlsVersion: '1.2'
+  healthCheckPath: '/healthz'
+  http20Enabled: true
+}
+
+@description('Optional. Kind of web app. Defaults to app.')
+@allowed([
+  'api'
+  'app'
+  'app,container,windows'
+  'app,linux'
+  'app,linux,container'
+  'functionapp'
+  'functionapp,linux'
+  'functionapp,linux,container'
+  'functionapp,linux,container,azurecontainerapps'
+  'functionapp,workflowapp'
+  'functionapp,workflowapp,linux'
+  'linux,api'
+])
+param kind string = 'app'
+
+@description('Optional. Diagnostic Settings for the App Service.')
+param appserviceDiagnosticSettings diagnosticSettingFullType[]
+
+@description('Optional. Diagnostic Settings for the App Service Plan.')
+param servicePlanDiagnosticSettings diagnosticSettingFullType[]
+
+@description('Optional. Diagnostic Settings for the ASE.')
+param aseDiagnosticSettings diagnosticSettingFullType[]
```

Comment:
I'm a bit confused 😄 

I resolved the comment where you stated that it would not make sense to expose the diagnostic settings, yet here we are 😄

Reply:
Same comment applies to other files

Reply:
I'm a bit confused - I had said it makes sense in my prior comment - "Thanks Alex! I implemented the interface the appropriate modules bc in this case, it does make sense that people should be able to configure the diagnostic settings in the way that they want, but the default is WAF aligned." I prob should've said "it makes sense" bc my brain almost inserted a not after does lol.

Reply:
Hot damn - ok that's my bad, sorry. At least we were confused together 🤜🔥🤛.
Resolving...

Reply:
No worries haha